Crash on `cmdx s jreleaser/jreleaser`

### aqua info

```console
$ aqua info

```


### Overview

`cmdx s jreleaser/jreleaser` crashes at the moment. I suppose the version of aqua involved is 2.45.0-8, and I guess it's because of this release which does not parse as a proper version: https://github.com/jreleaser/jreleaser/releases/tag/early-access

(I know there are workarounds, but perhaps it would be feasible to make aqua not crash.)

### How to reproduce

aqua.yaml

```yaml

```

Other related code such as local Registry

```

```

Executed command and output

```console
$ cmdx s -B jreleaser/jreleaser # in aqua-registry
```


### Expected behaviour

No crash, graceful error message.

### Actual behaviour

Crash, backtrace in details below.

<details>

```
$ cmdx s -B jreleaser/jreleaser
+ set -eu

PACKAGE=${PACKAGE#https://github.com/}

bash scripts/check_commands.sh
bash scripts/check_gpgsign.sh
bash scripts/check_diff.sh
bash scripts/checkout.sh "$PACKAGE"

if [ "$IS_RECREATE" = true ]; then
  aqua exec -- cmdx rm
fi

bash scripts/start.sh
bash scripts/scaffold.sh "$PACKAGE" "$CMD" "$LIMIT" "$CONFIG"
aqua exec -- cmdx gr
bash scripts/commit.sh "$PACKAGE"
bash scripts/test.sh "$PACKAGE"
bash scripts/start.sh aqua-registry-windows
bash scripts/test-windows.sh "$PACKAGE"

[INFO] Checking if the container aqua-registry exists
[INFO] Creating a container aqua-registry
[INFO] Get a GitHub Access token by gh auth token
...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x7a8e93]

goroutine 1 [running]:
github.com/hashicorp/go-version.(*Version).String(0x0)
	/home/runner/go/pkg/mod/github.com/hashicorp/go-version@v1.7.0/version.go:386 +0x33
github.com/hashicorp/go-version.(*Version).Compare(0xc000035a90, 0x0)
	/home/runner/go/pkg/mod/github.com/hashicorp/go-version@v1.7.0/version.go:119 +0x39
github.com/hashicorp/go-version.(*Version).LessThan(...)
	/home/runner/go/pkg/mod/github.com/hashicorp/go-version@v1.7.0/version.go:325
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.mergeFixedGroups.func1(0xc00086abd0?, 0xc0005a2690?)
	/home/runner/work/aqua/aqua/pkg/controller/generate-registry/group.go:172 +0x4f
sort.insertionSort_func({0xc0003b3f20?, 0xc0009395a0?}, 0x0, 0x5)
	/opt/hostedtoolcache/go/1.23.4/x64/src/sort/zsortfunc.go:12 +0xa7
sort.pdqsort_func({0xc0003b3f20?, 0xc0009395a0?}, 0xc00086abd0?, 0x0?, 0xc00086abd0?)
	/opt/hostedtoolcache/go/1.23.4/x64/src/sort/zsortfunc.go:73 +0x31b
sort.Slice({0xce3240?, 0xc00086abd0?}, 0xc0003b3f20)
	/opt/hostedtoolcache/go/1.23.4/x64/src/sort/slice.go:29 +0xc5
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.mergeFixedGroups({0xc0003da310, 0x5, 0xc0000645e0?})
	/home/runner/work/aqua/aqua/pkg/controller/generate-registry/group.go:171 +0x2d6
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.sortAndMergeGroups({0xc0004317a0, 0xe, 0xe57b60?})
	/home/runner/work/aqua/aqua/pkg/controller/generate-registry/group.go:195 +0x1d0
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).group(0xc000034dc0, 0xc000459b90, {0x7ffd0d994e0b, 0x13}, {0xc00047a140?, 0xe335c0?, 0xffffffffffffff01?})
	/home/runner/work/aqua/aqua/pkg/controller/generate-registry/group.go:281 +0x5b8
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).generatePackage(0xc000459b90?, 0xc0006b0480?, 0xc00043a508, {0x7ffd0d994e0b?, 0xc0003b4370?}, {0xc00047a140?, 0xc000596b68?, 0xc00010f94b?})
	/home/runner/work/aqua/aqua/pkg/controller/generate-registry/group.go:285 +0x31
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).getPackageInfoWithVersionOverrides(0xc000034dc0, {0x1035050, 0xc000397540}, 0xc000459b90, {0x7ffd0d994e0b, 0x13}, 0xc00043a508, 0x0?, 0xc0003b51d0)
	/home/runner/work/aqua/aqua/pkg/controller/generate-registry/version_overrides.go:135 +0x6bf
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).getPackageInfoMain(0xc000034dc0, {0x1035050, 0xc000397540}, 0xc000459b90, {0x7ffd0d994e0b?, 0x1?}, 0x0, 0xc0003b51d0)
	/home/runner/work/aqua/aqua/pkg/controller/generate-registry/generate.go:158 +0x62d
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).getPackageInfo(0x10?, {0x1035050?, 0xc000397540?}, 0x414fcb?, {0x7ffd0d994e0b?, 0xc000476d30?}, 0xc0003b5298, 0x46ce7d?)
	/home/runner/work/aqua/aqua/pkg/controller/generate-registry/generate.go:112 +0x49
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).genRegistry(0xc000034dc0, {0x1035050, 0xc000397540}, 0xc0003b5298, 0xc0003b51c8?, 0x410825?, {0x7ffd0d994e0b, 0x13})
	/home/runner/work/aqua/aqua/pkg/controller/generate-registry/generate.go:71 +0x65
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).GenerateRegistry(0xc000034dc0, {0x1035050, 0xc000397540}, 0xc0003b5298, 0xc000459b90, {0xc000476d30?, 0xc0003b53b0?, 0x10?})
	/home/runner/work/aqua/aqua/pkg/controller/generate-registry/generate.go:47 +0x1a5
github.com/aquaproj/aqua/v2/pkg/cli/genr.(*command).action(0xc000064500, 0xc0003979c0)
	/home/runner/work/aqua/aqua/pkg/cli/genr/command.go:145 +0x285
github.com/urfave/cli/v2.(*Command).Run(0xc000468160, 0xc0003979c0, {0xc000397a00, 0x4, 0x4})
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.27.5/command.go:276 +0x7e2
github.com/urfave/cli/v2.(*Command).Run(0xc000468840, 0xc000397640, {0xc0000ce000, 0x5, 0x5})
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.27.5/command.go:269 +0xa65
github.com/urfave/cli/v2.(*App).RunContext(0xc00041c800, {0x1035050, 0xc000397540}, {0xc0000ce000, 0x5, 0x5})
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.27.5/app.go:333 +0x5a5
github.com/aquaproj/aqua/v2/pkg/cli.Run({0x1035050, 0xc000397540}, 0xc000034d20, {0xc0000ce000, 0x5, 0x5})
	/home/runner/work/aqua/aqua/pkg/cli/runner.go:115 +0x90d
main.core(0xc000459b90, 0xc00004d220)
	/home/runner/work/aqua/aqua/cmd/aqua/main.go:45 +0x225
main.main()
	/home/runner/work/aqua/aqua/cmd/aqua/main.go:31 +0x4c
exit status 2
exit status 2
```

</details>

### Note

_No response_

--- a/pkg/controller/generate-registry/group.go
+++ b/pkg/controller/generate-registry/group.go
@@ -100,7 +100,9 @@ func mergeGroups(pkg *registry.PackageInfo, groups []*Group) []string { //nolint
 	}
 	pkg.VersionOverrides[len(pkg.VersionOverrides)-1].VersionConstraints = "true"
 	sort.Slice(releases, func(i, j int) bool {
-		return releases[i].Version.GreaterThan(releases[j].Version)
+		rI := releases[i]
+		rJ := releases[j]
+		return rI.GreaterThan(rJ)
 	})
 	versions := make([]string, len(releases))
 	for i, release := range releases {
@@ -169,7 +171,9 @@ func mergeFixedGroups(groups []*Group) []*Group {
 	}
 	arr := slices.Collect(maps.Values(m))
 	sort.Slice(arr, func(i, j int) bool {
-		return arr[i].releases[0].Version.LessThan(arr[j].releases[0].Version)
+		rI := arr[i].releases[0]
+		rJ := arr[j].releases[0]
+		return rI.LessThan(rJ)
 	})
 
 	// Move the group with NoAsset to the top.

--- a/pkg/controller/generate-registry/version_overrides.go
+++ b/pkg/controller/generate-registry/version_overrides.go
@@ -30,6 +30,20 @@ type Release struct {
 	assets        []*github.ReleaseAsset
 }
 
+func (r *Release) LessThan(r2 *Release) bool {
+	if r.Version != nil && r2.Version != nil {
+		return r.Version.LessThan(r2.Version)
+	}
+	return r.Tag < r2.Tag
+}
+
+func (r *Release) GreaterThan(r2 *Release) bool {
+	if r.Version != nil && r2.Version != nil {
+		return r.Version.GreaterThan(r2.Version)
+	}
+	return r.Tag > r2.Tag
+}
+
 func listPkgsFromVersions(pkgName string, versions []string) []*aqua.Package {
 	if len(versions) == 0 {
 		return nil

